tensorboard/data/server/data_compat.rs

```diff
@@ -45,6 +47,52 @@ pub enum EventValue {
     Summary(SummaryValue),
 }
 
+impl EventValue {
+    /// Consumes this event value and enriches it into a scalar.
+    ///
+    /// This supports `simple_value` (TF 1.x) summaries as well as non-empty tensors of type
+    /// `DT_FLOAT` or `DT_DOUBLE`. If a tensor has more than one item, the first will be taken.
```

Comment:
Sure, that works for me—happy to check here since this enrichment only
happens at commit time, not on every record.

I can go ahead and explicitly check the shape, too, and deny tensors of
size 1 but non-zero rank (e.g., `[[[0.25]]]`).